rust/processor/src/db_writer.rs

```diff
@@ -0,0 +1,309 @@
+        Error::BrokenTransactionManager => "BrokenTransactionManager",
+        _ => "Other",
+    }
+}
+
+// A holder struct for processors db writing so we don't need to keep adding new params
+#[derive(Clone)]
+pub struct DbWriter {
+    pub processor_name: &'static str,
+    pub db_pool: PgDbPool,
+    pub query_sender: AsyncSender<QueryGenerator>,
+    pub per_table_chunk_sizes: AHashMap<String, usize>,
+    pub skip_tables: AHashSet<String>,
+}
+
+impl DbWriter {
+    pub fn new(
+        processor_name: &'static str,
+        db_pool: PgDbPool,
+        query_sender: AsyncSender<QueryGenerator>,
+        per_table_chunk_sizes: AHashMap<String, usize>,
+        skip_tables: AHashSet<String>,
+    ) -> Self {
+        Self {
+            processor_name,
+            db_pool,
+            query_sender,
+            per_table_chunk_sizes,
+            skip_tables,
+        }
+    }
+
+    pub fn chunk_size<Item: field_count::FieldCount>(&self, table_name: &str) -> usize {
+        get_config_table_chunk_size::<Item>(table_name, &self.per_table_chunk_sizes)
+    }
+
+    pub async fn send_in_chunks<Item>(&self, table_name: &'static str, items_to_insert: Vec<Item>)
+    where
+        Item: field_count::FieldCount
+            + serde::Serialize
+            + for<'de> serde::Deserialize<'de>
+            + Send
+            + Sync
+            + Clone
+            + 'static,
+        Vec<Item>: DbExecutable,
+    {
+        // TODO: ideally this kind of skip is done much earlier in the process
+        if self.skip_tables.contains(table_name) {
+            return;
+        }
+
+        DB_EXECUTOR_CHANNEL_CHUNK_INSERT_COUNT
+            .with_label_values(&[self.processor_name, table_name])
+            .inc();
+
+        let chunk_size = self.chunk_size::<Item>(table_name);
+        let chunks = get_chunks(items_to_insert.len(), chunk_size);
+        for (start_ind, end_ind) in chunks {
+            let items = items_to_insert[start_ind..end_ind].to_vec();
+            self.query_sender
+                .send(QueryGenerator {
+                    table_name,
+                    db_executable: Box::new(items),
+                })
+                .await
+                .expect("Error sending query generator to db writer task");
+        }
+
+        DB_EXECUTOR_CHANNEL_SIZE
+            .with_label_values(&[self.processor_name])
+            .set(self.query_sender.len() as i64);
+    }
+}
+
+pub async fn launch_db_writer_tasks(
+    query_receiver: AsyncReceiver<QueryGenerator>,
+    processor_name: &'static str,
+    num_tasks: usize,
+    conn: PgDbPool,
+) {
+    info!(
+        processor_name = processor_name,
+        service_type = PROCESSOR_SERVICE_TYPE,
+        "[Parser] Starting db writer tasks",
+    );
+    let tasks = (0..num_tasks)
+        .map(|_| launch_db_writer_task(query_receiver.clone(), processor_name, conn.clone()))
+        .collect::<Vec<_>>();
+    futures::future::try_join_all(tasks)
```

Comment:
you probably want a separate runtime for this

Reply:
agreed, I probably do- will do those changes in a follow up PR